utils/core-modules/contracts/modules/AssociatedSystemsModule.sol

```diff
@@ -126,7 +126,10 @@ contract AssociatedSystemsModule is IAssociatedSystemsModule {
         AssociatedSystem.Data storage store = AssociatedSystem.load(id);
 
         if (store.proxy != address(0)) {
-            _upgradeToken(id, impl);
+            if (store.impl != impl) {
+                _upgradeToken(id, impl);
+            }
+            ITokenModule(store.proxy).initialize(name, symbol, decimals);
```

Comment:
Might be worth adding a comment here. We do we wanna call `initialize` even when we're not calling `_upgradeToken`?